### ISSUE TITLE
Port to material-ui@next

### ISSUE THREAD

**Issue by leMaik** [TeamWertarbyte/material-ui-chip-input#88]
Material-UI currently gets a full rewrite that is in alpha state. This might be the best time to start porting this component.

Unfortunately, there is no AutoComplete yet, but manually implement it may make the code less hacky, which is nice. :relaxed: 

There is no ETA, I'll start working on this on the _next_ branch. Feel free to submit PRs to speed up the process. :wink:

**Comment by umidbekkarimov** [comment]
Yay! Would you make it composable?

e.g:
```javascript
<ChipAutoComplete
   input={<MyCustomTextField {...searchText} />}
   chips={datasource.map((value) => <MyCustomChip  {...value} />)}
/>
```

**Comment by leMaik** [comment]
@umidbekkarimov Looks good to me. :+1: But I'd also like to have a second component with a slightly simpler usage (e.g. just specify string values). Pretty much like how material-ui@next allows you to either compose your own TextField or use their ready-to-use one.

**Comment by viteksafronov** [comment]
material-ui won't have an AutoComplete component at all. They suggest to use react-autosuggest instead.

**Comment by leMaik** [comment]
@viteksafronov Yes, already saw that and we already had to remove auto-complete support in another project (where auto-complete wasn't really a core feature).
Maybe we can make react-autosuggest optional so that it works without the dependency but supports auto-complete, if it is available. :thinking:

**Comment by umidbekkarimov** [comment]
I guess after another [bummer](https://github.com/TeamWertarbyte/material-ui-chip-input/pull/142) with material-ui/AutoComplete, we should start working in `react-autosuggest` direction.

**Comment by leMaik** [comment]
@umidbekkarimov Yes indeed, but only for material-ui@1. We'll keep using AutoComplete for 0.x. :wink:

**Comment by thupi** [comment]
[paypal/downshift](https://github.com/paypal/downshift) Might also be worth looking at :-) I found it way less problematic when it comes ot multiselect and so on :-)

**Comment by Wenzil** [comment]
Any progress on this?

**Comment by leMaik** [comment]
@Wenzil Not yet, but we recently started porting our other components to material-ui 1.0.0, so the port of this component will soon be started. You may create a PR if you want to help us porting it or get started. :+1:

**Comment by xiaoouLi** [comment]
@leMaik Do you have an estimate when you can finish it?
We are planning a new feature which uses material-ui v1.0.0, 
but we are using your material-ui-chip-input, which means that we cannot do the new feature
right now. If you guys can finish the port in a short run, we will wait for you to finish. 
Otherwise, we will go with the old material-ui for this feature....

Really appreciate your answer!
Lois

**Comment by leMaik** [comment]
@xiaoouLi We don't have a use-case for this component at the moment, so unfortunately, there isn't much time to port it. I would really appreciate anyone who needs this component to start porting it, so that everyone will benefit, though! :+1: 

You really shouldn't go back to 0.x for this feature. :wink:

**Comment by justinmasse** [comment]
I'm also very interested in this. Needed for an application I'm working on right now. Is there an ETA and a list of things needed to be done?

**Comment by leMaik** [comment]
@justinmasse There is no ETA, but I'm working on it in my spare time. :wink: Things that still need to be done: Port everything. Chips are already displayed and the input is kind-of-working, but auto-complete is still missing. I'll try to publish a first release before christmas. :santa: :gift:

**Comment by marcw** [comment]
@leMaik Can we help you in any way porting this to mui 1.0?

**Comment by leMaik** [comment]
@marcw I'm currently trying to get the input field beside of the chips instead of below them, but above the line. It works in the 0.x version by re-implementing an entire input field, but I'd like not to do that here. A solution for this would be really helpful! :+1:

**Comment by leMaik** [comment]
For those who asked what's left to be done: I added checkboxes for the progress above. :wink:

**Comment by leMaik** [comment]
I just published [v1.0.0-beta.1](https://github.com/TeamWertarbyte/material-ui-chip-input/releases/tag/v1.0.0-beta.1), a first beta of the ported component. :tada:

Auto-complete is still missing (that's why I'm leaving this issue opened), but I'll add that soon™.

**Comment by marcw** [comment]
@leMaik Amazing! This demo https://material-ui-next.com/demos/autocomplete/ leaves Autocomplete to the `react-suggest` library. This library don't handle multiple values though. I believe https://github.com/paypal/downshift can be used with multiple values.

**Comment by leMaik** [comment]
@marcw react-suggest seems to do what we need here :+1: The chip input handles multiple values on its own, we just need the auto complete functionality.

**Comment by marcw** [comment]
@leMaik It would be event better then as marmelab/admin-on-rest already depends on `react-suggest`. Less dependency, more love!

**Comment by fzaninotto** [comment]
Any progress on this one?

**Comment by leMaik** [comment]
@fzaninotto Yes, see the checkboxes above. :wink: Everything except auto-complete is already done and released. I'll port the auto-complete feature, I just need to have some free time to do that.
If you want, you can help out by implementing auto-completion with `react-suggest`. :wink:

**Comment by fzaninotto** [comment]
Thanks, I had already seen your work so far. Kudos for that! I was enquiring about the auto-complete implementation.

As for giving a hand, perhaps the work I've done on [admin-on-rest's AutoCompleteInput](https://github.com/marmelab/admin-on-rest/blob/next/packages/react-admin/src/mui/input/AutocompleteInput.js), which uses `react-autosuggest`, can serve as a base? I know I've spend about 2 days on the actual implementation, and it was a real pain ;)

Looking forward to integrating the newest version of your lib in admin-on-rest. Keep up the good work!

**Comment by fzaninotto** [comment]
Hi, and happy new year! Any progress on the autocomplete?

**Comment by leMaik** [comment]
@fzaninotto Nope, but we love pull requests, especially since so many people seem to need this feature. :open_mouth:

**Comment by fzaninotto** [comment]
@marcw do you feel like giving a hand there? It would help us include this component into react-admin...

**Comment by leMaik** [comment]
@fzaninotto We're not using this component at the moment at all, so unfortunately, there is not much time to add big features like this one. :disappointed: I'd really like to implement this, though.

**Comment by marcw** [comment]
@fzaninotto Yes I got some time this afternoon. I'll check this out.

**Comment by marcw** [comment]
@fzaninotto @leMaik @djhi I started checking but I'm struggling bridging everything and won't have time to finish anytime soon :(

**Comment by leMaik** [comment]
@marcw Maybe submit a PR so that we can build upon it? :)

**Comment by marcw** [comment]
@leMaik I wish I had something to push :(

**Comment by fzaninotto** [comment]
@quorak kudos!

**Comment by nsaubi** [comment]
up ? Any news on the port status ?

Need this comp (which is the best I found for now) for material v1 ! :)

**Comment by leMaik** [comment]
@nsaubi https://github.com/TeamWertarbyte/material-ui-chip-input/tree/next

**Comment by ricardovf** [comment]
Great work so far! Any progress on the Autocomplete?

**Comment by leMaik** [comment]
@nsaubi Thanks. :) Well, at least we have an example for autocomplete: https://github.com/TeamWertarbyte/material-ui-chip-input/pull/192

Since material-ui doesn't contain an autocomplete component anymore, and I don't want to force users to use one specific of the many autocomplete libraries out there, we'll have to add an API so that users can easily plug-in their favorite autocomplete library. PRs welcome :tada:

**Comment by milufranz08** [comment]
I haven't been able to see the example of using material-ui-chip-input and react-autosuggest. Could someone please post the link to the example code? Thank you!
